[Experimental] Add Product filters overlay settings

roykho opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes https://github.com/woocommerce/woocommerce/issues/47039

This PR adds overlay specific settings to the Product Filters block.

### Note

- ~~The overlay preview svg is not added yet. There is just a placeholder. Waiting for design on that.~~ Jarek said we can implement this in the later phase.
- The icon size is using the RangeControl component with some hacky CSS to make it align with the proposed design. This is so we don't need to build a custom component if we don't have to.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Test by developers only

Prereq:
- Ensure you have the WC Beta Tester Plugin installed and go to Tools > WCA Test Helper > Features. Toggle on the experimental-blocks option.

1. Go to the editor and navigate to Product Catalog template.
2. Add the Product Filters (Experimental) block to the editor.
3. Highlight the Product Filters (Experimental) block.
4. You will see some new inspector control options.
5. You should see Overlay with Never, Mobile and Always settings.
6. Selecting Mobile and Always will further expose more settings.
7. You will also see Button settings for displaying as Icon, Icon and label and Icon only.
8. The Icon and label and Icon settings will expose more settings. These two settings will expose an icon style and icon size.
9. Selecting Label only should hide the settings mentioned in step 8.

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [x] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->
This is an experimental block so no changelog is required.

</details>

roykho committed:
Add overlay settings

roykho committed:
Merge branch 'trunk' into product-filters-overlay-settings

roykho committed:
Tweak icon size slider

roykho committed:
Add overlay template part link

Commit:
Add changefile(s) from automation for the following project(s): woocommerce-blocks, woocommerce

roykho committed:
Merge branch 'trunk' into product-filters-overlay-settings

roykho committed:
Fix scss linting

roykho committed:
Fix php linting

roykho committed:
Remove preview

roykho committed:
Rename setting props to be more clear

roykho committed:
Remove unused import

roykho committed:
Merge branch 'trunk' into product-filters-overlay-settings

roykho commented:
@dinhtungdu @thealexandrelara would you mind giving this a review?

dinhtungdu commented on plugins/woocommerce-blocks/assets/js/blocks/product-filters/edit.tsx:
```diff
@@ -121,8 +137,179 @@ export const Edit = ( {}: BlockEditProps< ProductFiltersBlockAttributes > ) => {
+								label={ __( 'BUTTON', 'woocommerce' ) }
+								selected={ attributes.overlayButtonStyle }
+								onChange={ (
+									value: 'label-icon' | 'label' | 'icon'
```
We can put this type in `types.ts` and reference it here.

dinhtungdu commented on plugins/woocommerce-blocks/assets/js/blocks/product-filters/edit.tsx:
```diff
@@ -121,8 +137,179 @@ export const Edit = ( {}: BlockEditProps< ProductFiltersBlockAttributes > ) => {
+						</p>
+					) }
+					{ ( attributes.overlay === 'mobile' ||
+						attributes.overlay === 'always' ) && (
```
Just a preference and completely optional, but this can reduce to single condition. But using not condition is not as explicit and straight forward as yours.

```suggestion
					{ ( attributes.overlay !== 'never' ) && (
```

dinhtungdu commented on plugins/woocommerce-blocks/assets/js/blocks/product-filters/block.json:
```diff
@@ -42,7 +44,24 @@
+		},
+		"overlayButtonStyle": {
+			"type": "string",
+			"default": "icon"
```
Based on my interpretation from the design, the default should be `label-icon`, right? @jarekmorawski can you confirm?

dinhtungdu commented on plugins/woocommerce-blocks/assets/js/blocks/product-filters/block.json:
```diff
@@ -42,7 +44,24 @@
+		},
+		"overlayIconSize": {
+			"type": "number",
+			"default": "32"
```
Also, should the icon be default to this big? In the design, the icon size is 12px.

dinhtungdu commented on plugins/woocommerce-blocks/assets/js/blocks/product-filters/edit.tsx:
```diff
@@ -121,8 +137,179 @@ export const Edit = ( {}: BlockEditProps< ProductFiltersBlockAttributes > ) => {
+												| 'filter-icon-1'
+												| 'filter-icon-2'
+												| 'filter-icon-3'
+												| 'filter-icon-4'
```
Same as my other comment, this can also be put in `types.ts` and referenced here.

dinhtungdu commented on plugins/woocommerce-blocks/assets/js/blocks/product-filters/edit.tsx:
```diff
@@ -121,8 +137,179 @@ export const Edit = ( {}: BlockEditProps< ProductFiltersBlockAttributes > ) => {
+													size={ 32 }
+													icon={
+														<svg
+															xmlns="http://www.w3.org/2000/svg"
+															viewBox="0 0 24 24"
+															width="32"
+															height="32"
+															aria-hidden="true"
+														>
+															<path d="M10.541 4.20007H5.20245C4.27908 4.20007 3.84904 5.34461 4.54394 5.95265L10.541 11.2001V16.2001L10.541 17.9428C10.541 18.1042 10.619 18.2558 10.7504 18.3496L13.2504 20.1353C13.5813 20.3717 14.041 20.1352 14.041 19.7285V11.2001L19.3339 5.90718C19.9639 5.27722 19.5177 4.20007 18.6268 4.20007H13.041H10.541Z" />
+														</svg>
+													}
+												/>
+											}
+										/>
+										<ToggleGroupControlOption
+											value={ 'filter-icon-2' }
+											label={
+												<Icon
+													size={ 32 }
+													icon={
+														<svg
+															xmlns="http://www.w3.org/2000/svg"
+															viewBox="0 0 24 24"
+															width="32"
+															height="32"
+															aria-hidden="true"
+														>
+															<path
+																d="M4 11.5H8V10H4V11.5ZM0 0V1.5H12V0H0ZM2 6.5H10V5H2V6.5Z"
+																transform="translate(6,6)"
+															/>
+														</svg>
```
We should extract these custom icon into components, because we'll need to use them later for the editor preview and the saved content.

roykho replied:
So I have a dilemma here because I am finding several locations of where icons live. Some are here https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce/assets/images/blocks
Some are within the block directories themselves and some are here https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce-blocks/assets/js/icons/library

Also what is the naming convention we want? We can't just name an icon file filter.tsx because there could be many different filter icons that are used with different blocks. So we need to prefix (namespace) that right? But what is the convention?

I am happy to go with your suggestion here since you've been working in this for longer.

dinhtungdu replied:
> Some are here https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce/assets/images/blocks

This is place for images rather than icon, even images there are svg.

> Some are within the block directories themselves and some are here https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce-blocks/assets/js/icons/library

Let's move all icon to this folder and follow the naming convention there. We don't need a namespace because we import them from our own alias, so there won't be any conflict. We should move existing local icons too (but in other PRs).

dinhtungdu commented on plugins/woocommerce-blocks/assets/js/blocks/product-filters/editor.scss:
```diff
@@ -0,0 +1,46 @@
+$blue: #3858e9;
```
We should use color variable from Core here instead of only using fixed value.

```suggestion
$blue: var(--wp-components-color-accent, var(--wp-admin-theme-color, #3858e9));
```

dinhtungdu commented on plugins/woocommerce-blocks/assets/js/blocks/product-filters/types.ts:
```diff
@@ -1,7 +1,8 @@
+	overlay: string;
+	overlayIcon: string;
+	overlayButtonStyle: string;
+	overlayIconSize: number | undefined;
```
What about making this attribute optional?
```suggestion
	overlayIconSize?: number;
```

dinhtungdu commented on plugins/woocommerce/src/Blocks/BlockTypes/ProductFiltersOverlay.php:
```diff
@@ -36,4 +52,54 @@ protected function render( $attributes, $content, $block ) {
+		) {
+			$theme_slug = BlockTemplateUtils::theme_has_template_part( 'product-filters-overlay' ) ? wp_get_theme()->get_stylesheet() : BlockTemplateUtils::PLUGIN_SLUG;
+
+			if ( version_compare( get_bloginfo( 'version' ), '5.9', '<' ) ) {
```
Do we need to support older versions? We explicitly declared the minimum version is 6.4

```
 * Requires at least: 6.4
```

roykho replied:
So these were just copied over from the mini-cart for consistency. If we change it, we probably should update there too.

dinhtungdu replied:
I think we should update there too. Personally I won't trade unused code for consistency. So we have different preferences, let's include @thealexandrelara here :D. What do you think about this (and [this](https://github.com/woocommerce/woocommerce/pull/48732/files#r1656617941))?

dinhtungdu commented on plugins/woocommerce/src/Blocks/BlockTypes/ProductFiltersOverlay.php:
```diff
@@ -12,6 +17,17 @@ class ProductFiltersOverlay extends AbstractBlock {
 	 */
 	protected $block_name = 'product-filters-overlay';
 
+	/**
+	 * Constructor.
+	 *
+	 * @param AssetApi            $asset_api Instance of the asset API.
+	 * @param AssetDataRegistry   $asset_data_registry Instance of the asset data registry.
+	 * @param IntegrationRegistry $integration_registry Instance of the integration registry.
+	 */
+	public function __construct( AssetApi $asset_api, AssetDataRegistry $asset_data_registry, IntegrationRegistry $integration_registry ) {
+		parent::__construct( $asset_api, $asset_data_registry, $integration_registry, $this->block_name );
+	}
```
This constructor do nothing, it's not clear to me why we need it here. 🤔

roykho replied:
Same as previous comment, it was just copied from mini-cart.

dinhtungdu commented on plugins/woocommerce-blocks/assets/js/blocks/product-filters/edit.tsx:
```diff
@@ -121,8 +137,179 @@ export const Edit = ( {}: BlockEditProps< ProductFiltersBlockAttributes > ) => {
+						<>
+							<RadioControl
+								className="wc-block-editor-product-filters__overlay-button-style-toggle"
+								label={ __( 'BUTTON', 'woocommerce' ) }
```
The label case is transformed using CSS, so we don't have to use all uppercase here.

dinhtungdu replied:
![image](https://github.com/woocommerce/woocommerce/assets/5423135/dd67b3c7-a56d-46be-9841-3f92d80510e3)

dinhtungdu commented on plugins/woocommerce-blocks/assets/js/blocks/product-filters/edit.tsx:
```diff
@@ -121,8 +137,179 @@ export const Edit = ( {}: BlockEditProps< ProductFiltersBlockAttributes > ) => {
+	const templatePartEditUri = getSetting(
+		'templatePartProductFiltersOverlayEditUri',
+		''
+	) as string;
```
You can avoid type-casting by providing the type of returned value:
```suggestion
	const templatePartEditUri = getSetting< string >(
		'templatePartProductFiltersOverlayEditUri',
		''
	);
```

dinhtungdu replied:
Or you can also use `getSettingWithCoercion` for extra safety, but IMO, in this case, `getSetting` is enough.

dinhtungdu commented on plugins/woocommerce-blocks/assets/js/blocks/product-filters/edit.tsx:
```diff
@@ -5,17 +5,29 @@ import {
+	PanelBody,
+	RadioControl,
+	__experimentalToggleGroupControlOption as ToggleGroupControlOption,
+	__experimentalToggleGroupControl as ToggleGroupControl,
```
Just a common practice I see in our repo, we usually put experimental components at last.

dinhtungdu commented on plugins/woocommerce-blocks/assets/js/blocks/product-filters/edit.tsx:
```diff
@@ -121,8 +137,179 @@ export const Edit = ( {}: BlockEditProps< ProductFiltersBlockAttributes > ) => {
+						className="wc-block-editor-product-filters__overlay-toggle"
+						isBlock={ true }
+						value={ attributes.overlay }
+						onChange={ ( value: 'never' | 'mobile' | 'always' ) => {
```
This can also be moved to `types.ts`.

dinhtungdu commented on plugins/woocommerce-blocks/assets/js/blocks/product-filters/edit.tsx:
```diff
@@ -121,8 +137,179 @@ export const Edit = ( {}: BlockEditProps< ProductFiltersBlockAttributes > ) => {
+											}
+										/>
+									</ToggleGroupControl>
+									<div className="wc-block-editor-product-filters__overlay-button-size">
```
We can pass custom class direct to the `RangeControl` component.

dinhtungdu reviewed (commented):
Thanks for working on this PR!

> You will see some new inspector control options.
You should see Overlay with Never, Mobile and Always settings.
Selecting Mobile and Always will further expose more settings.
You will also see Button settings for displaying as Icon, Icon and label and Icon only.
The Icon and label and Icon settings will expose more settings. These two settings will expose an icon style and icon size.
Selecting Label only should hide the settings mentioned in step 8.

I can confirm the testing instructions applied well in my test, inspector settings are shown/hidden based on overlay setting and button style as expected.

In addition to review comments, I have a question related to the navigation button. I don't see it when the overlay setting is Always, does that belong to other PRs?

As @jarekmorawski mentioned:

> When Mobile is selected, we display a new section letting users customize the trigger button (the same as in the Navigation block).
Users can tweak the appearance of the trigger button (label and icon, label, icon).

> When Always is selected, we display a preview and a link to the overlay template part.
The block’s structure changes. We remove most of the inner blocks (including the filters, which are then pushed to the overlay template part) and replace them with a new Navigation inner block.
Users can decide on the button’s position (default or sticky) and customize its appearance (label style, width, colors, fonts, dimensions, etc).

So, there is a difference between Mobile and Always:

- Mobile: the navigation button settings belong to the Product Filters block.
- Always: the navigation button settings belong to a dedicated Navigation block.

![image](https://github.com/woocommerce/woocommerce/assets/5423135/762eecf7-dddd-430c-8e2e-10814d78cfe1)

![image](https://github.com/woocommerce/woocommerce/assets/5423135/c0dee84f-504b-47f3-84e6-fee238f365aa)

roykho committed:
Update plugins/woocommerce-blocks/assets/js/blocks/product-filters/editor.scss

Co-authored-by: Tung Du <dinhtungdu@gmail.com>

roykho committed:
Update plugins/woocommerce-blocks/assets/js/blocks/product-filters/edit.tsx

Co-authored-by: Tung Du <dinhtungdu@gmail.com>

roykho committed:
Update plugins/woocommerce-blocks/assets/js/blocks/product-filters/types.ts

Co-authored-by: Tung Du <dinhtungdu@gmail.com>

roykho committed:
Update plugins/woocommerce-blocks/assets/js/blocks/product-filters/edit.tsx

Co-authored-by: Tung Du <dinhtungdu@gmail.com>

roykho committed:
Move experimental controls to last of list

roykho committed:
Improve typescript code

thealexandrelara commented:
Thank you for working on this, really great to see the progress on this.

✅ Never, Mobile and Always settings are displayed correctly
✅ It's possible to select between Never, Mobile and Always options.

---

⚠️ When hovering over the Mobile and Always settings there is no block preview

![image](https://github.com/woocommerce/woocommerce/assets/20469356/a75158a2-d49c-4335-81da-ed9b7ee83e00)

Since you mentioned that Jarek is still working on the preview part, perhaps we can address this together with the preview that appears when the Mobile and Always options are selected.

---

❌ When the Always option is selected, the Button and Icon Size options should be hidden

| Current | Expected |
|--------|--------|
| ![CleanShot 2024-06-27 at 17 24 40@2x](https://github.com/woocommerce/woocommerce/assets/20469356/32eaeec1-740c-4c46-9b4b-85ded05d5e01) | ![image](https://github.com/woocommerce/woocommerce/assets/20469356/6bc943ec-d603-48e9-a81d-c4d895d7c912) | 

---

❌ When the Always option is selected, the individual filters should be hidden and only a dedicated Filters button should appear
Tung mentioned in his comment above, but there is a different in the way the block is displayed after selecting the Always option.

| Current | Expected |
|--------|--------|
| ![CleanShot 2024-06-27 at 17 53 09@2x](https://github.com/woocommerce/woocommerce/assets/20469356/473821d6-2b8b-4952-a42c-c7ca927f7475) | ![image](https://github.com/woocommerce/woocommerce/assets/20469356/080ab0f2-b3d6-45e5-bc30-d4aea103bc4d) | 

---
❌ When the Mobile option is selected, the Apply button should appear at the bottom

![CleanShot 2024-06-27 at 17 55 52](https://github.com/woocommerce/woocommerce/assets/20469356/0b6a1867-b65a-4f4a-b8fb-e88725a6af76)

roykho commented:
@thealexandrelara - Thanks for the review.

> When the Always option is selected, the Button and Icon Size options should be hidden

I believe that is working as designed. In Figma, I asked this question to Jarek and I concluded that Mobile and Always works almost the same. In my mind that means both options will have the buttons except in Mobile, either the desktop viewport shrinks down to mobile size or it is displayed on a mobile device. Please let me know if you had another idea of this. And if we have conflicting ideas, it means these options are confusing and users would likely to be confused as well.

dinhtungdu reviewed (commented):
> ❌ When the Mobile option is selected, the Apply button should appear at the bottom

@thealexandrelara from the design, I can see the Apply button when the Overlay is set to Never too, not only Mobile.

<details><summary>Design</summary>

![image](https://github.com/woocommerce/woocommerce/assets/5423135/201b472b-023e-4913-b297-16b2dceb8a49)


</details>

> > When the Always option is selected, the Button and Icon Size options should be hidden

> I believe that is working as designed. In Figma, I asked this question to Jarek and I concluded that Mobile and Always works almost the same. In my mind that means both options will have the buttons except in Mobile, either the desktop viewport shrinks down to mobile size or it is displayed on a mobile device. Please let me know if you had another idea of this. And if we have conflicting ideas, it means these options are confusing and users would likely to be confused as well.

@roykho I have the same [observation](https://github.com/woocommerce/woocommerce/pull/48732#pullrequestreview-2144456011) as @thealexandrelara. Can you or @jarekmorawski share more insights on this one?

roykho commented:
@dinhtungdu 

> We don't need a namespace because we import them from our own alias

I don't think you understood what I meant. What if we have two icons both name filter.tsx?? Those two filter icons could look different in different context. So you do need a namespace. In this case it can be `product-filters-overlay-filter.tsx` or something to that affect.

> Personally I won't trade unused code for consistency. So we have different preferences

I don't mean we should do it this way. I mean since it was working with the mini-cart, I just copied the code over to make that part work. I didn't look too much into it because I know we might change this part at the end.

> I have the same https://github.com/woocommerce/woocommerce/pull/48732#pullrequestreview-2144456011 as @thealexandrelara.

How do you see it working? What does `Always` mean to you?

dinhtungdu commented:
@roykho 

> I don't think you understood what I meant. What if we have two icons both name filter.tsx?? 

Yeah, I didn't understand but I think I get it now. I don't think we have any convention now. So what name to use is considered case by case, and we're here to discuss.

For the two icons in your component, IMO, we can name them `funnel` and `funnelAlt` respectively. The latter one can be `funnel-line`, or even `funnel-hamburger`, hah. I would try to avoid sticking an icon to a specific block/component and give them a generic name instead, as an icon can be used in more places in the future.

About the namespace you mentioned, I don't have any good idea which namespace to use and the mechanism we should follow. 

> I didn't look too much into it because I know we might change this part at the end.

I see, but we can also forget it and leave those unused codes in our codebase. So my preference is only adding what is necessary.

> How do you see it working? What does Always mean to you?

As shared my observation in my [initial review](https://github.com/woocommerce/woocommerce/pull/48732#pullrequestreview-2144456011):

> So, there is a difference between Mobile and Always:
>
> - Mobile: the navigation button settings belong to the Product Filters block.
> - Always: the navigation button settings belong to a dedicated Navigation block.
>
> <details><summary>Designs</summary>
> <p>
>
> ![image](https://github.com/woocommerce/woocommerce/assets/5423135/762eecf7-dddd-430c-8e2e-10814d78cfe1)
> 
> ![image](https://github.com/woocommerce/woocommerce/assets/5423135/c0dee84f-504b-47f3-84e6-fee238f365aa)
> </p>
> </details> 

I want to add more details to the comparison:

- Mobile: the navigation button settings belong to the Product Filters block. The navigation button is hidden in the editor.
- Always: the navigation button settings belong to a dedicated Navigation block and are hidden from the Product Filters block setting. The navigation button is visible in the editor preview, and filter heading, and all options are hidden.

roykho committed:
Remove unneeded constructor code

roykho committed:
Extract icons into own component

roykho committed:
Moved overlay template part link to the bottom

roykho committed:
Add apply button

roykho committed:
Merge branch 'trunk' into product-filters-overlay-settings

roykho committed:
Hide overlay settings when in always mode

roykho committed:
Remove overlay template part from list and put into general area

roykho committed:
Update api version to 3

roykho committed:
Merge branch 'trunk' into product-filters-overlay-settings

roykho commented:
So I've made updates to this PR based on the review feedback. I just have this one where I am still not 100% sure.

> Always: the navigation button settings belong to a dedicated Navigation block.

Does that mean we should create another block for this? Does this overlap with the Overlay Navigation block? How will users change the settings for the navigation block within the Product Filters block?

roykho committed:
Merge branch 'trunk' into product-filters-overlay-settings

dinhtungdu commented:
From what I understanding the design:

> Does that mean we should create another block for this?

Yes.

> Does this overlap with the Overlay Navigation block?

The functionality should be the same except the icon selector.

> How will users change the settings for the navigation block within the Product Filters block?

When the overlay setting is Always, interact with the Navigation block to change the setting.
When the overlay setting is Mobile, the navigation block settings are included in the Product Filters block, so users should change the navigation setting there.

It's best to ask @jarekmorawski for clarification.

jarekmorawski commented:
I can see how these interactions can be confusing and if you have any ideas how to make the experience better, I'll welcome them. The relationship between the overlay toggle and the navigation is tricky, but I'll try to explain it as best as I can:

**Never**

- The overlay template part isn't utilized. Users can manage all filters in the filters template part.
- There is no Navigation block.

**Mobile**

- The overlay template part is only shown on mobile and other small devices.
- There is no Navigation block; its features are surfaced in the Product Filters block.
- We would do it differently if the editor had a mobile-specific view. For now, we must somehow let users customize the button that will open the overlay on mobile
- The design in Figma still displays the fill version of the block because we don't have a way to resize it to show the mobile view. 

**Always**

- The overlay template part is always displayed when shoppers interact with the Navigation block on the front-end, both on desktop and mobile.
- There is no Product Filters block in the canvas anymore. It stays in sync with the overlay template part it's removed from the canvas and block structure. We only show the Navigation block.
- The overlay becomes the single source of truth for available filters; editing the overlay is the only way for users to add and remove filters.

Let me know if that clarifies things. We operate in a grey area since much of this functionality will appear in the core editor one day. For now, we must figure out the best path forward within the constraints.

roykho committed:
Merge branch 'trunk' into product-filters-overlay-settings

roykho commented:
@thealexandrelara @dinhtungdu What do you think about me opening a separate PR to add the navigation?

roykho committed:
Merge branch 'trunk' into product-filters-overlay-settings

roykho committed:
Merge branch 'trunk' into product-filters-overlay-settings

dinhtungdu commented on plugins/woocommerce/src/Blocks/BlockTypes/ProductFiltersOverlay.php:
```diff
@@ -30,10 +33,49 @@ protected function get_block_type_style() {
+			$site_editor_uri = add_query_arg(
+				array(
+					'canvas' => 'edit',
+					'path'   => '/template-parts/single',
```
I got redirected to this URL after clicking on the edit link: `http://woo.test/wp-admin/site-editor.php?canvas=edit&postId=woocommerce/woocommerce//product-filters-overlay&postType=wp_template_part`. I'm using WP 6.4 which is the minimum supported WP version.

roykho replied:
Yes that should be correct. It takes you to the overlay template part. I got the same in both WP 6.4 and 6.5.

dinhtungdu commented on plugins/woocommerce-blocks/assets/js/blocks/product-filters/edit.tsx:
```diff
@@ -62,6 +75,30 @@ const TEMPLATE: InnerBlockTemplate[] = [
 			heading: __( 'Rating', 'woocommerce' ),
 		},
 	],
+	[
+		'core/buttons',
+		{ layout: { type: 'flex' } },
+		[
+			[
+				'core/button',
+				{
+					text: __( 'Apply', 'woocommerce' ),
+					className: 'wc-block-product-filters__apply-button',
+					style: {
+						border: {
+							width: '0px',
+							style: 'none',
+						},
+						typography: {
+							textDecoration: 'none',
+						},
+						outline: 'none',
+						fontSize: 'medium',
+					},
+				},
+			],
+		],
+	],
```
Not sure about the Apply button as I think it doesn't relate to overlay setting as commented [here](https://github.com/woocommerce/woocommerce/pull/48732#pullrequestreview-2146953604). But we can change its behavior later so let's keep it for now.

dinhtungdu commented on plugins/woocommerce-blocks/assets/js/blocks/product-filters/edit.tsx:
```diff
@@ -128,8 +169,147 @@ export const Edit = ( {}: BlockEditProps< ProductFiltersBlockAttributes > ) => {
+							) }
+						</>
+					) }
+					<p className="wc-block-editor-product-filters__overlay-link">
+						<ExternalLink href={ templatePartEditUri }>
+							{ __( 'Edit overlay', 'woocommerce' ) }
+						</ExternalLink>
+					</p>
```
This paragraph is styled as a button. Should we use another tag for it. Also, only the link is clickable, not the "button", so it may be confusing to users.

dinhtungdu replied:
![image](https://github.com/user-attachments/assets/2a59e491-d3f7-422f-9a95-6d91f14dcf90)

dinhtungdu reviewed (approved):
Left some minor comments, otherwise, this is LGTM, so I'm pre-approving.

thealexandrelara commented on plugins/woocommerce-blocks/assets/js/icons/library/filter-three-lines.tsx:
```diff
@@ -0,0 +1,15 @@
+/**
+ * External dependencies
+ */
+import { SVG } from '@wordpress/primitives';
+
+const filterThreeLines = (
+	<SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
+		<path
+			d="M4 11.5H8V10H4V11.5ZM0 0V1.5H12V0H0ZM2 6.5H10V5H2V6.5Z"
+			transform="translate(6,6)"
+		/>
+	</SVG>
```
Should we import `Path` from the `@wordpress/primitives` library?

```suggestion
import { SVG, Path } from '@wordpress/primitives';

const filterThreeLines = (
	<SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
		<Path
			d="M4 11.5H8V10H4V11.5ZM0 0V1.5H12V0H0ZM2 6.5H10V5H2V6.5Z"
			transform="translate(6,6)"
		/>
	</SVG>
```

thealexandrelara commented on plugins/woocommerce-blocks/assets/js/icons/library/filter.tsx:
```diff
@@ -0,0 +1,12 @@
+/**
+ * External dependencies
+ */
+import { SVG } from '@wordpress/primitives';
+
+const filter = (
+	<SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
+		<path d="M10.541 4.20007H5.20245C4.27908 4.20007 3.84904 5.34461 4.54394 5.95265L10.541 11.2001V16.2001L10.541 17.9428C10.541 18.1042 10.619 18.2558 10.7504 18.3496L13.2504 20.1353C13.5813 20.3717 14.041 20.1352 14.041 19.7285V11.2001L19.3339 5.90718C19.9639 5.27722 19.5177 4.20007 18.6268 4.20007H13.041H10.541Z" />
+	</SVG>
+);
```
Should we use `Path` from `@wordpress/primitives` in here as well?

```suggestion
import { SVG, Path } from '@wordpress/primitives';

const filter = (
	<SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
		<Path d="M10.541 4.20007H5.20245C4.27908 4.20007 3.84904 5.34461 4.54394 5.95265L10.541 11.2001V16.2001L10.541 17.9428C10.541 18.1042 10.619 18.2558 10.7504 18.3496L13.2504 20.1353C13.5813 20.3717 14.041 20.1352 14.041 19.7285V11.2001L19.3339 5.90718C19.9639 5.27722 19.5177 4.20007 18.6268 4.20007H13.041H10.541Z" />
	</SVG>
);
```

thealexandrelara reviewed (changes requested):
Thank you for working on this! Here are the results of my testing:

✅ The controls to customize the layout and overlay are being displayed.
✅ In the Overlay settings, when the “Never” option is selected, no other settings are displayed.
✅ In the Overlay settings, when the “Mobile” option is selected, some controls to customize the Icon and Button are being displayed.
⚠️  In the Overlay settings, when the “Mobile” option is selected, the Edit overlay button is displayed. However, only the link is clickable, not the button itself.
❌ Currently, there are no E2E tests covering these new features. Would you mind to include some E2E tests?

roykho committed:
Update plugins/woocommerce-blocks/assets/js/icons/library/filter.tsx

Co-authored-by: Alexandre Lara <allexandrelara@gmail.com>

roykho committed:
Update plugins/woocommerce-blocks/assets/js/icons/library/filter-three-lines.tsx

Co-authored-by: Alexandre Lara <allexandrelara@gmail.com>

roykho committed:
Fix overlay button link style and clickable

roykho committed:
Remove edit overlay link from never mode

roykho committed:
Add e2e tests for the filters overlay settings

roykho commented:
@dinhtungdu @thealexandrelara Thanks for the review. I've pushed up changes based on the review.

- Updated the styling of the "Edit overlay" link so the entire button is clickable
- Added e2e tests to cover all the different combination of settings.

roykho committed:
Merge branch 'trunk' into product-filters-overlay-settings

thealexandrelara reviewed (approved):
Thank you for working on this. Great work! 🚀
